internal/storage/factory/factory.go

```diff
@@ -0,0 +1,56 @@
+package factory
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/xo/dburl"
```

Comment:
So how do we feel about this dburl package? It seems decent to me. But if you don't want another dep, I can always try the ole "Windsurf, implement a minimal version of the dburl package and inline it here" approach. In the other direction, dburl does have a unified way of evening making the db connection to delete even this code. It's all based on this go stdlib [`database/sql` interface](https://pkg.go.dev/database/sql#DB) which we could probably be using instead of any of our custom backend stuff. Hey, I didn't know about it either ha.

Reply:
I let Windsurf take a stab at it #12

Reply:
I think #12 is promising, we can roll with that.